instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/rules/assertions/MetricAssert.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.rules.assertions;
+
+import static io.opentelemetry.instrumentation.jmx.rules.assertions.DataPointAttributes.attributeGroup;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.metrics.v1.Metric;
+import io.opentelemetry.proto.metrics.v1.NumberDataPoint;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.internal.Integers;
+import org.assertj.core.internal.Iterables;
+import org.assertj.core.internal.Objects;
+
+public class MetricAssert extends AbstractAssert<MetricAssert, Metric> {
+
+  private static final Objects objects = Objects.instance();
+  private static final Iterables iterables = Iterables.instance();
+  private static final Integers integers = Integers.instance();
+
+  private boolean strict;
+
+  private boolean descriptionChecked;
+  private boolean unitChecked;
+  private boolean typeChecked;
+  private boolean dataPointAttributesChecked;
+
+  MetricAssert(Metric actual) {
+    super(actual, MetricAssert.class);
+  }
+
+  public void setStrict(boolean strict) {
+    this.strict = strict;
+  }
+
+  public void strictCheck() {
+    strictCheck("description", /* expectedCheckStatus= */ true, descriptionChecked);
+    strictCheck("unit", /* expectedCheckStatus= */ true, unitChecked);
+    strictCheck("type", /* expectedCheckStatus= */ true, typeChecked);
+    strictCheck(
+        "data point attributes", /* expectedCheckStatus= */ true, dataPointAttributesChecked);
```

Comment:
[for reviewer] this ensures that the test explicitly tests all the common properties of JMX metrics, which could be a feature that would be removed if we used `io.opentelemetry.sdk.testing.assertj.MetricAssert`, however this also means some duplicated effort so it could make sense to reuse what is available.